### ISSUE TITLE
pylab import replaced by 'py' when matplotlib is not installed but py is

### ISSUE THREAD

**Issue by chaen** [mgedmin/findimports#10]
Hi,

The `pylab` import is seen as a `py` module

Create a file with just that import
```
[chaen]$ cat /tmp/test.py 
from pylab import hist
```

Run findimports on it

```
[chaen]$ findimports /tmp/test.py 
test:
  py
```

The installed version is 1.5.1 

```
[chaen]$ pip freeze | grep findimports
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
findimports==1.5.1
```

Any idea what is going wrong ? 
Thanks !

**Comment by mgedmin** [comment]
Oh, this is interesting!

I cannot reproduce it:

```
$ virtualenv -p python2.7 /tmp/py27

$ /tmp/py27/bin/pip install findimports
Successfully installed findimports-1.5.1

$ cat /tmp/test.py
from pylab import hist

$ /tmp/py27/bin/findimports /tmp/test.py
/tmp/test.py: could not find pylab.hist
test:
  pylab.hist

$ /tmp/py27/bin/pip install matplotlib
Successfully installed backports.functools-lru-cache-1.5 cycler-0.10.0 kiwisolver-1.1.0 matplotlib-2.2.4 numpy-1.16.5 pyparsing-2.4.2 python-dateutil-2.8.0 pytz-2019.3 six-1.12.0 subprocess32-3.5.4

$ /tmp/py27/bin/findimports /tmp/test.py
test:
  pylab
```

Can you tell me what `findimports /tmp/test.py -d` would print?  For me it looks like

```
digraph ModuleDependencies {
  node[shape=box];
  mod0[label="test"];
  node[style=dotted];
  extmod0[label="pylab"];
  mod0 -> extmod0;
}
```

and the reason I ask is I want to make sure it's not the plaintext output getting truncated midword (although I've no idea why it would be truncated midword).

Another test would be to find out what Python interpreter is used for findimports with

```
$ head -n 1 $(which findimports)
```

and then run that interpreter and in it try

```
>>> import pylab
>>> pylab.__file__
```

which for me produces '/tmp/py27/local/lib/python2.7/site-packages/pylab.pyc'.


Actually, I have a Theory.  Bear with me a moment.

**Comment by mgedmin** [comment]
Reproduced!

- virtualenv /tmp/py
- /tmp/py/bin/pip install py findimports
- /tmp/py/bin/findimports /tmp/test.py

produces the same output as your code.

And the reason is that in this loop:
https://github.com/mgedmin/findimports/blob/caa0e71f2d87fdfc09fb4999f02780d2f4649214/findimports.py#L502-L509

I'm trying to take off the last part of the dotted name with

            name = name[:name.rfind('.')]

and name.rfind('.') returns -1 when there are no dots in it, which cuts off the last character, one by one, until pylab becomes py, which exists in your site-packages!

```

**Comment by mgedmin** [comment]
(In my defense I originally wrote this code in 2003.  Python 2.3 was the modern version.  str.rindex ... existed, so I've no excuse!)

**Comment by mgedmin** [comment]
Released findimports 1.5.2 with the fix.

**Comment by chaen** [comment]
Woh, thanks a lot, impressive response speed :-)
